### ISSUE TITLE
Usage of non standard event.path in DropContainer

### ISSUE THREAD

**Issue by AntoineDuComptoirDesPharmacies** [grommet/grommet#6426]
### Expected Behavior

When clicking outside of Drop/Select Drop with a custom `containerTarget`, the drop should close itself.

### Actual Behavior

When clicking outside of Drop/Select Drop with a custom `containerTarget`, the application crash on Firefox.

This issue seems to come from this fix : https://github.com/grommet/grommet/pull/6120
The code used the field `path` which is non standard.

```
TypeError
event.path is undefined
onClickDocument
https://z9939s.csb.app/node_modules/grommet/es6/components/Drop/DropContainer.js:211:85
EventListener.handleEvent*$csb$eval/DropContainer</<
https://z9939s.csb.app/node_modules/grommet/es6/components/Drop/DropContainer.js:229:16
commitHookEffectListMount
https://z9939s.csb.app/node_modules/react-dom/cjs/react-dom.development.js:23049:26
commitPassiveMountOnFiber
https://z9939s.csb.app/node_modules/react-dom/cjs/react-dom.development.js:24816:38
commitPassiveMountEffects_complete
https://z9939s.csb.app/node_modules/react-dom/cjs/react-dom.development.js:24781:34
commitPassiveMountEffects_begin
https://z9939s.csb.app/node_modules/react-dom/cjs/react-dom.development.js:24768:41
commitPassiveMountEffects
https://z9939s.csb.app/node_modules/react-dom/cjs/react-dom.development.js:24756:34
flushPassiveEffectsImpl
https://z9939s.csb.app/node_modules/react-dom/cjs/react-dom.development.js:26990:28
flushPassiveEffects
https://z9939s.csb.app/node_modules/react-dom/cjs/react-dom.development.js:26935:14
commitRootImpl/<
https://z9939s.csb.app/node_modules/react-dom/cjs/react-dom.development.js:26725:9
workLoop
https://z9939s.csb.app/node_modules/scheduler/cjs/scheduler.development.js:266:42
flushWork
https://z9939s.csb.app/node_modules/scheduler/cjs/scheduler.development.js:239:14
performWorkUntilDeadline
https://z9939s.csb.app/node_modules/scheduler/cjs/scheduler.development.js:533:21
EventHandlerNonNull*$csb$eval/<
https://z9939s.csb.app/node_modules/scheduler/cjs/scheduler.development.js:571:3
$csb$eval
https://z9939s.csb.app/node_modules/scheduler/cjs/scheduler.development.js:633:5
H
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:100039
evaluate
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:112792
evaluateTranspiledModule
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:123173
c
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:112406
$csb$eval
https://z9939s.csb.app/node_modules/scheduler/index.js:6:20
H
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:100039
evaluate
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:112792
evaluateTranspiledModule
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:123173
c
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:112406
$csb$eval/<
https://z9939s.csb.app/node_modules/react-dom/cjs/react-dom.development.js:27:24
$csb$eval
https://z9939s.csb.app/node_modules/react-dom/cjs/react-dom.development.js:29769:5
H
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:100039
evaluate
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:112792
evaluateTranspiledModule
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:123173
c
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:112406
$csb$eval
https://z9939s.csb.app/node_modules/react-dom/index.js:37:20
H
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:100039
evaluate
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:112792
evaluateTranspiledModule
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:123173
c
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:112406
$csb$eval
https://z9939s.csb.app/node_modules/react-dom/client.js:3:16
H
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:100039
evaluate
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:112792
evaluateTranspiledModule
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:123173
c
https://codesandbox.io/static/js/sandbox.2e5f0d928.js:1:112406
$csb$eval
/index.js:2

  1 | import React, { useRef, useState } from "react";
> 2 | import { createRoot } from "react-dom/client";
  3 | 
  4 | import {
  5 |   grommet,

View compiled
▶ 2 stack frames were collapsed.
```

### URL, screen shot, or Codepen exhibiting the issue

https://codesandbox.io/s/grommet-v2-template-forked-z9939s?file=/index.js

### Steps to Reproduce

1. Use Firefox
2. Go to https://z9939s.csb.app/
3. Click on the Select to open the drop
4. Click outside the drop

### Your Environment

- Grommet version: 2.27.0
- Browser Name and version: Firefox 105.0.3
- Operating System and version (desktop or mobile): Windows 10

**Comment by AnshumanPadiya** [comment]
I believe we can make it conditional and check what the browser is providing, it works perfectly fine on chrome but as stated in the issue it's breaking on firefox a better way to solve this is to use [composedPath](https://developer.mozilla.org/en-US/docs/Web/API/Event/composedPath) as per my understanding. However, I am open to more suggestions. I would definitely like to work on this issue if no one is working on it, can someone please assign this to me?

**Comment by jcfilben** [comment]
Hi @AnshumanPadiya thanks for volunteering to work on this issue! I will add you as an assignee on it. Feel free to reach out if you have questions. Based on a quick google search, I think your `composedPath` idea is the right direction to explore.

**Comment by jcfilben** [comment]
Fixed in #6450
